### PR TITLE
Mkdocs

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
This PR introduces [Mkdocs](https://github.com/mkdocs/mkdocs/) for generating documentation website from markdown.

Here's what the site would look like:

https://christopherchudzicki.github.io/mitx-grading-library/

Deploying the documentation site to github pages is astonishingly simple. To deploy for the first time or to update existing documentation:

```bash
> cd mitxgraders
> mkdocs gh-deploy
```

Edit: To view documentation locally,

```bash
> cd mitxgraders
> mkdocs serve
```

Need to:
- [x] fix broken links
- [x] look into deploying automatically on push 
    **update:** well, I've looked into it anyway... Not sure the best approach. Discuss below.

### CODE REVIEW

**Comment by ChristopherChudzicki** [comment]
So, this was not strictly necessary for writing MatrixGrader documentation. But I find reading the documentation much easier now, and `mkdocs` is soooo easy.

**Comment by jolyonb** [comment]
Ah, I see you want to fix some broken links first. Good idea :-) Probably want to add a link appropriately too (via `README.md`). Let's get this uploading to the mitodl site first, and then we'll worry about configuring autoupdate.

**Comment by ChristopherChudzicki** [comment]
> You still want to generate those markdown files from docstrings at some point though, right?

Upon further reflection, I decided that the lack of syntax-highlighting in module-level docstrings was going to drive me nuts. 

I've just learned that you can run doctest on markdown files (e.g., `python -m doctest somefile.md`).

The downside is that examples would have to be doctests, which means an annoying abundance of `>>>` and `...`. I don't really mind *writing* those, but I've always found them annoying when trying to copy-paste from documentation.

Anyway, I might try the doctest approach for examples in MatrixGrader's documentation, but not in this PR.

**Comment by jolyonb** [comment]
Huh, didn't realize you could doctest markdown files... alrighty then. Let's keep that in mind. Let me know when this is ready.

**Comment by ChristopherChudzicki** [comment]
@jolyonb I consider this ready for merge:

- I've fixed the broken links
- Added a github button in upper right
- some other minor tweaks

I'm not sure if it's worth trying to integrate this into travis:
- I think it would be very easy to make travis re-deploy the documentation on every build... But we only want to re-deploy the documentation after merging a PR with master.
- I think it would be possible to re-deploy docs automatically after merges by:
    - using [Github Releases](https://help.github.com/articles/creating-releases/)
    - setting travis to [recognize the releases as deployment](https://docs.travis-ci.com/user/deployment/releases/), then adding an `after_deploy` command.

But I really don't think it's worth the effort. Simple enough to just run `mkdocs gh-deploy` locally after a merge. (Or just before a merge—it doesn't matter. `mkdocs gh-deploy` pushes directly to the `gh-pages` branch, not to `master`.)

I could look into hosting with something other than github pages, but `mkdocs gh-deploy` is soooo easy.

**Comment by ChristopherChudzicki** [comment]
Oh, and I already deployed to

https://mitodl.github.io/mitx-grading-library/

**Comment by jolyonb** [comment]
This is looking really good. Once the unicode issue is cleared up, good by me.

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    - Sampling Sets: 'grading_math/sampling.md'
+    - Function List: 'grading_math/function_list.md'
+    - AsciiMath Renderer Definitions: 'grading_math/renderer.md'
+  - Grading Lists:
+    - ListGrader: 'grading_lists/list_grader.md'
+    - SingleListGrader: 'grading_lists/single_list_grader.md'
+  - Changelog: 'changelog.md'
+markdown_extensions:
+    - toc:
+        permalink: 
```
[the remark this answers is not in the document]
Comment (ChristopherChudzicki) [reply mkdocs.yml]: The value of `permalink` controls what displays on hover. The default (`permalink: true`) is a paragraph symbol:
screen shot 2018-07-30 at 4 29 20 pm

That box (`permalink: `) displays a nicer link:
screen shot 2018-07-30 at 4 29 33 pm

But... I really have no idea what is up what the box . I copied that part of the configuration straight from the MkDocs docs:

https://github.com/mkdocs/mkdocs/blob/master/mkdocs.yml#L35

I can't figure out a nice way to get the same link symbol. We can just use the default paragraph symbol if you want.

[the remark this answers is not in the document]
Comment (jolyonb) [reply mkdocs.yml]: Hmm. Ok, I do like the linky thing. I'm a little worried about a text file containing a unicode character without a character encoding specified. Any chance we can use the unicode codepoint instead? https://stackoverflow.com/questions/35746109/how-to-do-unicode-escaping-in-yaml-multiline-string If that doesn't work, then I'm happy to just stick with the unicode character.

[the remark this answers is not in the document]
Comment (ChristopherChudzicki) [reply mkdocs.yml]: Good idea. Done!

**Comment by jolyonb** [comment]
I'll assume you've pushed this and it still works :-)

**Pull request merged**